lib/client.js

```diff
@@ -130,6 +132,13 @@ Client.prototype.request = function(opts, res) {
         req.write(opts.body);
     req.end();
 
+    if (this.config.timeout){
+        req.setTimeout(this.config.timeout, function() {
+          req.abort();
+          res(new Error('connection timed out'));
```

Comment:
I think this ends up invoking the callback twice -- aborting the request will also separately invoke the response due to the error handler.